syncano/models/options.py

```diff
@@ -133,15 +133,23 @@ def get_endpoint_methods(self, name):
         endpoint = self.get_endpoint(name)
         return endpoint['methods']
 
-    def resolve_endpoint(self, name, properties):
-        endpoint = self.get_endpoint(name)
+    def resolve_endpoint(self, endpoint_name, properties, http_method=None):
+        if http_method and not self.is_http_method_available_for_endpoint(http_method, endpoint_name):
+            raise SyncanoValidationError(
+                'HTTP method {0} not allowed for endpoint name "{1}".'.format(http_method, endpoint_name)
+            )
+        endpoint = self.get_endpoint(endpoint_name)
 
-        for name in endpoint['properties']:
-            if name not in properties:
-                raise SyncanoValueError('Request property "{0}" is required.'.format(name))
+        for endpoint_name in endpoint['properties']:
+            if endpoint_name not in properties:
+                raise SyncanoValueError('Request property "{0}" is required.'.format(endpoint_name))
 
         return endpoint['path'].format(**properties)
 
+    def is_http_method_available_for_endpoint(self, http_method_name, endpoint_name):
```

Comment:
`is_http_method_available_for_endpoint` -> `is_http_method_available` is sufficient